File: secondExercise_2/src/main/java/Screen.java
public class Screen {
    int sequence;
    Movie movie;
    int startTime;
    int endTime;
    double fee;
    DiscountPolicy discountPolicy;

    public Screen(int sequence, Movie movie, int startTime, int endTime, double fee, DiscountPolicy discountPolicy) {
        this.sequence = sequence;
        this.movie = movie;
        this.startTime = startTime;
        this.endTime = endTime;
        this.fee = fee;
        this.discountPolicy = discountPolicy;
    }

    public double getFee() {
        return fee;
    }

    public double calculateFee() {
        return this.fee - discountPolicy.calculateDiscountFee(this);
    }
}

File: secondExercise/src/main/kotlin/Screen.kt
class Screen(
    val sequence: Int,
    val movie: Movie,
    val startTime: Int,
    val endTime: Int,
    val fee: Double,
    val discountPolicy: DiscountPolicy
) {
    fun calculateFee(): Double {

        return this.fee - this.discountPolicy.calculateDiscountFee(this)
    }

    fun isBetweenTime(start: Int, end: Int): Boolean = this.startTime in (start + 1) until end
}
